src/python/pants/backend/go/dependency_inference.py

```diff
@@ -4,13 +4,14 @@
 
 from dataclasses import dataclass
 
+from pants.backend.go.util_rules.context import GoBuildContext
 from pants.build_graph.address import Address
 from pants.engine.environment import EnvironmentName
 from pants.engine.unions import union
 from pants.util.frozendict import FrozenDict
 
 
-@union(in_scope_types=[EnvironmentName])
+@union(in_scope_types=[EnvironmentName, GoBuildContext])
```

Comment:
Tried adding this to solve the rule graph issue. Still rule graph issue. Also, tried adding it to all other `@union` uses but still have rule graph errors.